be/src/exec/pipeline/sink/sink_io_buffer.h

```diff
@@ -84,10 +84,21 @@ class SinkIOBuffer {
     virtual void cancel_one_sinker() { _is_cancelled = true; }
 
     virtual void close(RuntimeState* state) {
-        _is_finished = true;
         if (_exec_queue_id != nullptr) {
-            bthread::execution_queue_stop(*_exec_queue_id);
+            int ret = bthread::execution_queue_stop(*_exec_queue_id);
+            if (ret != 0) {
+                auto error_msg = fmt::format("Fail to stop execution queue: {}", std::strerror(errno));
+                LOG(WARNING) << error_msg;
+            }
+
+            ret = bthread::execution_queue_join(*_exec_queue_id); // make sure the stop task has been executed
```

Comment:
will this interface block and wait until it finishes?

Reply:
yes